ec/ecresource/organizationresource/mapper_to_model.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func apiToModel(ctx context.Context, member models.OrganizationMembership, invitationPending bool, diagnostics *diag.Diagnostics) *OrganizationMember {
+	organizationRole := organizationRoleApiToModel(member)
+	if diagnostics.HasError() {
+		return nil
+	}
+
+	deploymentRoles := deploymentRolesApiToModel(ctx, member, diagnostics)
+	if diagnostics.HasError() {
+		return nil
+	}
+
+	projectElasticsearchRoles := elasticsearchRolesApiToModel(ctx, member, diagnostics)
+	if diagnostics.HasError() {
+		return nil
+	}
+
+	projectObservabilityRoles := observabilityRolesApiToModel(ctx, member, diagnostics)
+	if diagnostics.HasError() {
+		return nil
+	}
+
+	projectSecurityRoles := securityRolesApiToModel(ctx, member, diagnostics)
+	if diagnostics.HasError() {
+		return nil
+	}
+
+	return &OrganizationMember{
+		Email:                     types.StringValue(member.Email),
+		InvitationPending:         types.BoolValue(invitationPending),
+		UserID:                    types.StringValue(nilToEmpty(member.UserID)),
+		OrganizationRole:          organizationRole,
+		DeploymentRoles:           *deploymentRoles,
+		ProjectElasticsearchRoles: *projectElasticsearchRoles,
+		ProjectObservabilityRoles: *projectObservabilityRoles,
+		ProjectSecurityRoles:      *projectSecurityRoles,
+	}
+}
+
+func nilToEmpty(id *string) string {
+	if id == nil {
+		return ""
+	}
+	return *id
+}
+
+func organizationRoleApiToModel(member models.OrganizationMembership) types.String {
+	if member.RoleAssignments != nil &&
+		member.RoleAssignments.Organization != nil &&
+		len(member.RoleAssignments.Organization) > 0 &&
+		member.RoleAssignments.Organization[0] != nil &&
+		member.RoleAssignments.Organization[0].RoleID != nil {
+		id := member.RoleAssignments.Organization[0].RoleID
+		return types.StringValue(*id)
+	} else {
+		return types.StringNull()
+	}
+}
+
+func deploymentRolesApiToModel(ctx context.Context, member models.OrganizationMembership, diagnostics *diag.Diagnostics) *types.Set {
+	var result []DeploymentRoleAssignment
+	if member.RoleAssignments != nil {
+		for _, roleAssignment := range member.RoleAssignments.Deployment {
+			deploymentIds, diags := types.SetValueFrom(ctx, types.StringType, roleAssignment.DeploymentIds)
+			if diags.HasError() {
+				diagnostics.Append(diags...)
+				return nil
+			}
+			applicationRoles, diags := types.SetValueFrom(ctx, types.StringType, roleAssignment.ApplicationRoles)
+			if diags.HasError() {
+				diagnostics.Append(diags...)
+				return nil
+			}
+
+			result = append(result, DeploymentRoleAssignment{
+				Role:              types.StringValue(roleApiToModel(*roleAssignment.RoleID, Deployment)),
+				ForAllDeployments: forAllApiToModel(roleAssignment.All),
+				DeploymentIDs:     deploymentIds,
+				ApplicationRoles:  applicationRoles,
+			})
+		}
+	}
+	roleAssignments, diags := types.SetValueFrom(ctx, setElementObjectType(deploymentRoleAssignmentsSchema()), result)
```

Comment:
nit-pick: maybe change it to this?
```suggestion
	roleAssignments, diags := types.SetValueFrom(ctx, deploymentRoleAssignmentsSchema().NestedObject.GetAttributes().Type(), result)
```